File: Medium/JavaScript/2171. Removing Minimum Number of Magic Beans.js
var minimumRemoval = function(beans) {
    // Initialize values to store max and sum
    // Then stort the array
    let max = 0;
    let sum = 0;
    beans.sort((a, b) => a - b);

    // Use a for loop to remove the beans from the bags with
    // the least amount of beans in them until each non-empty
    // bag has the same amount of beans
    for(let i = 0; i < beans.length; i++){
        sum += beans[i];
        max = Math.max(max, beans[i] * (beans.length - i));
    }

    // Return the amount of beans that had to be removed
    return sum - max;
};

File: Medium/Java/2171. Removing Minimum Number of Magic Beans.java
class Solution {
    public long minimumRemoval(int[] beans) {
        // Initialize long values to store max and sum
        // Then sort the array
        long max = 0;
        long sum = 0;
        Arrays.sort(beans);
        
        // Use a for loop to remove the beans from the bags with
        // the least amount of beans in them until each non-empty
        // bag has the same number of beans
        for(int i = 0; i < beans.length; i++){
            sum += beans[i];
            max = Math.max(max, (long) beans[i] * (beans.length - i));
        }

        // Return the amount of beans that had to be removed
        return sum - max;
    }
}
